// @sentry/react/dist/index.js
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var browser_1 = require("@sentry/browser");
/**
 * A global side effect that makes sure Sentry events that user
 * `@sentry/react` will correctly have Sentry events associated
 * with it.
 */
function createReactEventProcessor() {
    if (browser_1.addGlobalEventProcessor) {
        browser_1.addGlobalEventProcessor(function (event) {
            event.sdk = tslib_1.__assign(tslib_1.__assign({}, event.sdk), { name: 'sentry.javascript.react', packages: tslib_1.__spread(((event.sdk && event.sdk.packages) || []), [
                    {
                        name: 'npm:@sentry/react',
                        version: browser_1.SDK_VERSION,
                    },
                ]), version: browser_1.SDK_VERSION });
            return event;
        });
    }
}
tslib_1.__exportStar(require("@sentry/browser"), exports);
var profiler_1 = require("./profiler");
Object.defineProperty(exports, "Profiler", { enumerable: true, get: function () { return profiler_1.Profiler; } });
Object.defineProperty(exports, "withProfiler", { enumerable: true, get: function () { return profiler_1.withProfiler; } });
Object.defineProperty(exports, "useProfiler", { enumerable: true, get: function () { return profiler_1.useProfiler; } });
var errorboundary_1 = require("./errorboundary");
Object.defineProperty(exports, "ErrorBoundary", { enumerable: true, get: function () { return errorboundary_1.ErrorBoundary; } });
Object.defineProperty(exports, "withErrorBoundary", { enumerable: true, get: function () { return errorboundary_1.withErrorBoundary; } });
var redux_1 = require("./redux");
Object.defineProperty(exports, "createReduxEnhancer", { enumerable: true, get: function () { return redux_1.createReduxEnhancer; } });
var reactrouterv3_1 = require("./reactrouterv3");
Object.defineProperty(exports, "reactRouterV3Instrumentation", { enumerable: true, get: function () { return reactrouterv3_1.reactRouterV3Instrumentation; } });
var reactrouter_1 = require("./reactrouter");
Object.defineProperty(exports, "reactRouterV4Instrumentation", { enumerable: true, get: function () { return reactrouter_1.reactRouterV4Instrumentation; } });
Object.defineProperty(exports, "reactRouterV5Instrumentation", { enumerable: true, get: function () { return reactrouter_1.reactRouterV5Instrumentation; } });
Object.defineProperty(exports, "withSentryRouting", { enumerable: true, get: function () { return reactrouter_1.withSentryRouting; } });
createReactEventProcessor();
